File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/uncomment/direct/Java/Go/s910447421_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"sort"
)

var ans int
var balls [][]int
var N int

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)

	N = nextInt(scanner)
	ans = N
	balls = make([][]int, N)
	for i := 0; i < N; i++ {
		x := nextInt(scanner)
		y := nextInt(scanner)
		balls[i] = []int{x, y}
	}

	sort.Slice(balls, func(i, j int) bool {
		return balls[i][0] < balls[j][0]
	})
	solve()

	for i := range balls {
		balls[i][0], balls[i][1] = balls[i][1], balls[i][0]
	}

	sort.Slice(balls, func(i, j int) bool {
		return balls[i][0] < balls[j][0]
	})
	solve()

	fmt.Println(ans)
}

func solve() {
	for add := 1; add < N; add++ {
		p := balls[add][0] - balls[0][0]
		q := balls[add][1] - balls[0][1]
		current := check(p, q)
		if current < ans {
			ans = current
		}
	}
}

func check(p, q int) int {
	taken := make([]bool, N)
	ret := 0
	for {
		added := false
		var start int
		var x, y int
		for i := 0; i < N; i++ {
			if !taken[i] {
				added = true
				ret++
				start = i
				taken[i] = true
				x = balls[i][0] + p
				y = balls[i][1] + q
				break
			}
		}
		if !added {
			break
		}
		for i := start + 1; i < N; i++ {
			if balls[i][0] == x && balls[i][1] == y {
				taken[i] = true
				x += p
				y += q
			}
		}
	}
	return ret
}

func nextInt(scanner *bufio.Scanner) int {
	scanner.Scan()
	num, _ := strconv.Atoi(scanner.Text())
	return num
}

// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/comment/uncomment/direct/Java/Python/s910447421_commented.py
import sys
import math

class Main:
    def __init__(self):
        self.ans = 100
        self.balls = []
        self.N = 0
        self.taken = []

    def main(self):
        input = sys.stdin.read().split()
        index = 0
        self.N = int(input[index])
        index += 1
        self.ans = self.N
        self.balls = [[0, 0] for _ in range(self.N)]
        for i in range(self.N):
            self.balls[i][0] = int(input[index])
            index += 1
            self.balls[i][1] = int(input[index])
            index += 1
        self.balls.sort(key=lambda x: x[0])
        self.solve()
        for i in range(self.N):
            x = self.balls[i][0]
            y = self.balls[i][1]
            self.balls[i][0] = y
            self.balls[i][1] = x
        self.balls.sort(key=lambda x: x[0])
        self.solve()
        print(self.ans)

    def solve(self):
        for add in range(1, self.N):
            self.taken = [False] * self.N
            p = self.balls[add][0] - self.balls[0][0]
            q = self.balls[add][1] - self.balls[0][1]
            self.ans = min(self.ans, self.check(p, q))

    def check(self, p, q):
        ret = 0
        while True:
            added = False
            start = 0
            x = -1
            y = -1
            for i in range(self.N):
                if not self.taken[i]:
                    added = True
                    ret += 1
                    start = i
                    self.taken[i] = True
                    x = self.balls[i][0] + p
                    y = self.balls[i][1] + q
                    break
            if not added:
                break
            for i in range(start + 1, self.N):
                if self.balls[i][0] == x and self.balls[i][1] == y:
                    self.taken[i] = True
                    x += p
                    y += q
        return ret

if __name__ == "__main__":
    main_instance = Main()
    main_instance.main()

# End of Code
